services/schemas/soldier.zschema.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from 'zod';
+
+const soldier = z.object({
+  id: z.string(),
+  rank: z.string(),
```

Comment:
rank should be on of some types. What if I try to insert
 `{id: 0, rank: "pag", name: "Evyatar", limitations: ["literally everything"],
 duties: [] }`
 ?

Reply:
anak